### PR TITLE
Implementation of  `machinery config foo=bar` syntax

### LINKED ISSUE #1001: `machinery config` should accept `=` when setting values
Right now `machinery config foo bar` is used to set the key `foo` to value `bar`.

`machinery config foo=bar` should be additionally supported.

### PR DESCRIPTION
**Pull request by vwallfahrer**
Fixed #1001

### CODE REVIEW

**Comment by tgoettlicher** [comment]
The CLI doesn't complain if you provide too many args:
`./bin/machinery config hints=true superfluous`

File: lib/cli.rb [hunk 1]
```diff
@@ -717,14 +717,20 @@ def self.parse_scopes(scope_string)
   arg "VALUE", :optional
   command "config" do |c|
     c.action do |global_options,options,args|
-      key = args[0]
-      value = args[1]
+      if !args[0].nil? && args[0].include?("=")
```
Comment (thardeck) [inline lib/cli.rb]: Instead of `!args[0].nil?` you can check for `args[0]`.

File: lib/cli.rb [hunk 2]
```diff
@@ -717,14 +717,20 @@ def self.parse_scopes(scope_string)
...
+        parameter = args[0].split("=")
+        key = parameter[0]
+        value = parameter[1]
```
Comment (thardeck) [inline lib/cli.rb]: I would use
```
key, value = args[0].split("=")
```

File: lib/local_system.rb [hunk 3]
```diff
@@ -52,7 +52,7 @@ def validate_machinery_compatibility
 It still could work very well. If you run into issues or would like to provide us feedback, you are welcome to file an issue at https://github.com/SUSE/machinery/issues/new or write an email to machinery@lists.suse.com.
 Oficially supported operating systems are: '#{supported_oses}'
 
-To disable this message in the machinery configuration use 'machinery config perform-support-check false'
+To disable this message in the machinery configuration use 'machinery config perform-support-check=false'
```
Comment (thardeck) [inline lib/local_system.rb]: Can you replace `machinery` by `#{$0}` so it gets extended. It is not directly related to your pr but if you change the line anyway.

File: spec/unit/cli_spec.rb [hunk 4]
```diff
@@ -559,6 +559,12 @@
         with("foo", "bar")
       run_command(["config", "foo", "bar"])
     end
+
+    it "handles parameter with equal syntax" do
```
Comment (thardeck) [inline spec/unit/cli_spec.rb]: s/equal/equal sign/

**Comment by thardeck** [comment]
lgtm

**Pull request closed without merging**